crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -86,85 +83,151 @@ impl gossip::GossipEventHandler for Space {
         &mut self,
     ) -> gossip::GossipEventHandlerResult<Vec<Arc<KitsuneAgent>>> {
         // while full-sync this is just a clone of list_by_basis
-        let res = self.agents.keys().cloned().collect();
-        Ok(async move { Ok(res) }.boxed().into())
+        let all_agents = self
+            .agents
+            .keys()
+            .cloned()
+            .map(|agent| {
+                self.evt_sender
+                    .query_agent_info_signed(QueryAgentInfoSignedEvt {
```

Comment:
at the moment we've decided to go with a single peer store per space - so repeating this call for each agent is going to return duplicate data.

Reply:
Also, while technically any agents in the self.agents should also have published AgentInfoSigned entries... for testing purposes where QueryAgentInfoSigned just returns an empty vec![] should we make sure the self.agent agents are in the results too?